test/e2e/sor.spec.ts

```diff
@@ -0,0 +1,135 @@
+/** Tests fetching common token trades from the SOR endpoint, converting them into a transaction, and simulates with Tenderly to make sure they work*/
+require('dotenv').config();
+
+import { SorRequest } from '../../src/types';
+import { parseFixed } from '@ethersproject/bignumber';
+import { BigNumber } from 'ethers';
+import { testSorRequest } from '../lib/sor';
+import { Network } from '../../src/constants/general';
+import { getInfuraUrl } from '../../src/utils';
+import { forkSetup, getBalances, setEthBalance, setTokenBalance } from '../lib/helpers';
+import { JsonRpcProvider } from '@ethersproject/providers';
+import { TOKENS } from '../../src/constants/addresses';
+
+const WALLET_ADDRESS =
+  process.env.WALLET_ADDRESS || '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266';
+
+jest.unmock('@ethersproject/contracts');
+jest.unmock('@balancer-labs/sdk');
+jest.setTimeout(30000);
+
+let provider, signer; 
+
+const hardhatUrl = process.env.HARDHAT_URL || `http://127.0.0.1:8545`;
+const rpcUrl = process.env.RPC_URL || getInfuraUrl(Network.MAINNET);
+
+if (!rpcUrl) {
+  console.error('Env variable RPC_URL or INFURA_PROJECT_ID must be set to run these tests')
+  process.exit(1);
+}
+
+/**
+ * These tests do the following:
+ * - Make a request to the /sor endpoint
+ * - Convert the response into a swap transaction
+ * - Simulate that transaction with Hardhat and ensure it completes correctly
+ */
+describe('SOR Endpoint E2E tests', () => {
+  beforeEach(async () => {
+    provider = new JsonRpcProvider(hardhatUrl, Network.MAINNET);
+    await forkSetup(provider.getSigner(), rpcUrl);
+  });
+
+  describe('Mainnet Tests', () => {
+    beforeEach(async () => {
+      await provider.send('hardhat_impersonateAccount', [WALLET_ADDRESS]);
+      signer = await provider.getSigner(WALLET_ADDRESS);
+      await setEthBalance(signer, parseFixed('100', 18));
+    });
+
+    it('Should be able to sell DAI for BAL', async () => {
+      const { DAI, BAL } = TOKENS[Network.MAINNET];
+      const sorRequest: SorRequest = {
+        sellToken: DAI.address,
+        buyToken: BAL.address,
+        orderKind: 'sell',
+        amount: parseFixed('100', 18).toString(),
+        gasPrice: BigNumber.from('0x174876e800').toString(),
+      };
+      await setTokenBalance(signer, DAI, sorRequest.amount)
+      const balances = await getBalances(signer, [BAL]);
+      await testSorRequest(signer, Network.MAINNET, sorRequest);
+      const newBalances = await getBalances(signer, [BAL]);
+      expect(BigNumber.from(newBalances.BAL).gt(balances.BAL));
```

Comment:
Should we not be testing that the amount received is a specific amount? The problem we had with routing through boosted pools (described in the asana ticket) was that the SOR was returning incorrect values. Will these tests failif that were to happen again?

Reply:
It's really checking if the batch swap fails. Which should happen if the numbers are bad. Though I could add a sanity check test that makes sure it fails when the numbers are off.

Reply:
I'd added a test that purposely fails. It also revealed that my limit settings were wrong which is good.